setup.py

```diff
@@ -1,7 +1,14 @@
 #!/usr/bin/env python
+import sys
 
 from setuptools import setup
 
+if sys.version_info < (3, 4):
+    # functools.singledispatch is in stdlib from Python 3.4 onwards.
+    extra_requires = ['singledispatch>=3.4']
+else:
+    extra_requires = []
+
```

Comment:
I think this will give us issues in creating wheels - we'll need different wheels for Python 2, Python 3 < 3.4 and Python 3 >= 3.4 due to different requirements, which is a bit annoying, but possible, as per [PEP 425](https://www.python.org/dev/peps/pep-0425/).